internal/clearingway/menu.go

```diff
@@ -179,6 +181,270 @@ func (c *Clearingway) MenuStaticRespond(s *discordgo.Session, i *discordgo.Inter
+		return
+	}
+
+	menu := g.Menus.Menus[menuName]
+	additionalData := menu.AdditionalData
+
+	userOptions := i.MessageComponentData().Values
+	userOptionsMap := make(map[string]struct{})
+	for _, opt := range userOptions {
+		userOptionsMap[opt] = struct{}{}
+	}
+
+	userRoles := i.Member.Roles
+	userRolesMap := make(map[string]struct{})
+	for _, roleId := range userRoles {
+		userRolesMap[roleId] = struct{}{}
+	}
+
+	// check which roles to add or remove
+	rolesToAdd := []*MenuRoleHelper{}
+	rolesToRemove := []*Role{}
+	for _, roleHelper := range additionalData.Roles {
+		role := roleHelper.Role
+
+		_, selected := userOptionsMap[role.DiscordRole.ID]
+		_, present := userRolesMap[role.DiscordRole.ID]
+
+		if selected && !present {
+			rolesToAdd = append(rolesToAdd, roleHelper)
+		} else if !selected && present {
+			rolesToRemove = append(rolesToRemove, role)
+		}
+	}
+
+	successRoles := []string{}
+	removedRoles := []string{}
+	failedRoles := []string{}
+	errorRoles := []string{}
+
+	// add/remove roles based on prereq met
+	for _, roleHelper := range rolesToAdd {
+		prereq := roleHelper.Prerequisite
+		if roleHelper.Prerequisite != nil {
+			if _, prereqMet := userRolesMap[prereq.DiscordRole.ID]; !prereqMet {
+				failedRoles = append(failedRoles, roleHelper.Prerequisite.DiscordRole.ID)
+				continue
+			}
+		}
+
+		err := roleHelper.Role.AddToCharacter(i.GuildID, i.Member.User.ID, s)
+		if err != nil {
+			fmt.Printf("Error adding role: %+v\n", err)
+			errorRoles = append(errorRoles, roleHelper.Role.DiscordRole.ID)
+			continue
+		}
+
+		successRoles = append(successRoles, roleHelper.Role.DiscordRole.ID)
+	}
+
+	for _, role := range rolesToRemove {
+		err := role.RemoveFromCharacter(i.GuildID, i.Member.User.ID, s)
+		if err != nil {
+			fmt.Printf("Error removing role: %+v\n", err)
+			errorRoles = append(errorRoles, role.DiscordRole.ID)
+			continue
+		}
+
+		removedRoles = append(removedRoles, role.DiscordRole.ID)
+	}
+
+	// form response string
+	responseMsg := ""
+
+	for i, roles := range [2][]string{successRoles, removedRoles} {
+		verb := ""
+		if i == 0 {
+			verb = "added"
+		} else {
+			verb = "removed"
+		}
+
+		length := len(roles)
+		if length == 1 {
+			responseMsg += fmt.Sprintf("Successfully %v role: <@&%v>\n", verb, roles[0])
+		} else {
+			responseMsg += fmt.Sprintf("Successfully %v roles: ", verb)
+			for _, role := range roles {
+				responseMsg += fmt.Sprintf("<@&%v> ", role)
+			}
+			responseMsg += "\n"
+		}
+	}
+
+	for _, failedRole := range failedRoles {
+		responseMsg += fmt.Sprintf("You do not have the required role: <@&%v>\n", failedRole)
+	}
+
+	if len(errorRoles) != 0 {
+		responseMsg += "An error has occurred. Please contact an admin with this message.\nRole actions that failed: "
+		for _, role := range errorRoles {
+			responseMsg += fmt.Sprintf("<@&%v> ", role)
+		}
+	}
+
+	if len(responseMsg) == 0 {
+		responseMsg = "Nothing has been done!"
```

Comment:
if all the other response strings have new lines, we should include it within this string too